src/js/program/ad-program-controller.js

```diff
@@ -75,6 +75,7 @@ export default class AdProgramController extends ProgramController {
         this.provider = null;
         return super.setActiveItem(index)
             .then((mediaController) => {
+                this.trigger(AD_MEDIA_LOADED);
```

Comment:
Why not just use `instream.on('playlistItem')` ? This is vast only and essentially the same thing (vast is already handling this for ad pods).

Reply:
`this.instreamPlayer.on('playlistItem', this.playlistItemHandler, this);`

Reply:
I guess this includes the provider and media-controller being setup which happens after that. If that's the case and what you need then LGTM.